File: postm/entities/post.py
from dataclasses import dataclass

@dataclass
class Post(object):
    id: str
    title: str
    image: str | None
    description: str
    createdAt: str
    updatedAt: str

    def toJson(self) -> dict[str, str]:
        return {
            "id": self.id,
            "title": self.title,
            "image": self.image,
            "description": self.description,
            "createdAt": self.createdAt,
            "updatedAt": self.updatedAt
        }

File: src/entities/post.ts
export default class User {
    
    readonly id: string;
    public image: string | null;
    public title: string;
    public description: string;

    public constructor(id: string, image: string | null, title: string, description: string) {
        this.id = id;
        this.image = image;
        this.title = title;
        this.description = description;
    }

}